packages/modules/devices/shelly/device.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+import logging
+from typing import List, Iterable
+
+from helpermodules.cli import run_using_positional_cli_args
+from modules.common.abstract_device import DeviceDescriptor
+from modules.common.configurable_device import (ConfigurableDevice, ComponentFactoryByType, MultiComponentUpdater,
+                                                SingleComponentUpdateContext)
+from modules.devices.shelly.inverter import ShellyInverter
+from modules.devices.shelly.config import Shelly, ShellyConfiguration
+from modules.devices.shelly.config import ShellyInverterSetup, ShellyInverterConfiguration
+
+
+log = logging.getLogger(__name__)
+
+
+def create_device(device_config: Shelly) -> ConfigurableDevice:
+    def create_inverter_component(component_config: ShellyInverterSetup) -> ShellyInverter:
+        return ShellyInverter(device_config.id, component_config, device_config.configuration.address)
+
+    def update_components(components: Iterable[ShellyInverter]) -> None:
+        for component in components:
+            with SingleComponentUpdateContext(component.component_info):
+                component.update()
+
+    return ConfigurableDevice(
+        device_config=device_config,
+        component_factory=ComponentFactoryByType(
+            inverter=create_inverter_component
+        ),
+        component_updater=MultiComponentUpdater(update_components)
+    )
```

Comment:
```suggestion
    return ConfigurableDevice(
        device_config=device_config,
        component_factory=ComponentFactoryByType(
            inverter=create_inverter_component
        ),
        component_updater=IndependentComponentUpdater(lambda component: component.update())
    )
```
Der Multicomponent-Updater wird hier nicht gebraucht, da jede Komponente separat abgefragt wird.

Reply:
geändert